src/api/console_stringifier.py

```diff
@@ -33,16 +34,61 @@ def _modify_unit(self, unit: str) -> str:
         """
         Returns the modified unit.
         """
-        unit = (
-            unit.replace(r"\squared", "^2")
-            .replace(r"\cubed", "^3")
-            .replace("\\per\\", "/")
-            .replace(r"\per", "/")
-            .replace("\\", " ")
-            .strip()
-        )
-
-        if unit[0] == "/":
-            unit = f"1{unit}"
+
+        # Detect "\squared" etc.:
+        unit = unit.replace(r"\squared", "^2").replace(r"\cubed", "^3")
+        unit = re.sub(r"(\s+)\^", "^", unit)
+
+        # Detect special units:
+        unit = unit.replace(r"\percent", r"\%").replace(r"\degree", "°")
+
+        # Detect "/":
+        unit = unit.replace("/", " / ")
+
+        # Iterate over unit parts:
+        unit_parts = re.split(r"\\|\s", unit)
+        numerator_parts = []
+        denominator_parts = []
+        is_next_part_in_denominator = False
+
+        for unit_part in unit_parts:
+            # Skip empty parts:
+            if unit_part == "":
+                continue
+
+            # If next part is a denominator part:
+            if unit_part in ("/", "per"):
+                is_next_part_in_denominator = True
+                continue
+
+            # Add part to numerator or denominator:
+            if is_next_part_in_denominator:
+                denominator_parts.append(unit_part)
+                is_next_part_in_denominator = False
+            else:
+                numerator_parts.append(unit_part)
+
+        # Assemble unit:
+        unit = ""
```

Comment:
Please do developers a favor and use a new variable here instead of redefining the purpose of an old one (and one that was even passed in as parameter)

Reply:
Done.